static/js/libs/channels.test.js

```diff
@@ -0,0 +1,533 @@
+        latest: {
+          name: "latest",
+          risks: {
+            stable: {
+              name: "stable",
+              branches: {
+                _base: {
+                  name: "_base"
+                }
+              }
+            },
+            edge: {
+              name: "edge",
+              branches: {
+                _base: {
+                  name: "_base"
+                }
+              }
+            }
+          }
+        },
+        test: {
+          name: "test",
+          risks: {
+            candidate: {
+              name: "candidate",
+              branches: {
+                _base: {
+                  name: "_base"
+                }
+              }
+            }
+          }
+        }
+      });
+    });
+  });
+
+  describe("risk/branch", () => {
+    beforeEach(() => {
+      channelList = ["stable", "beta/test", "edge/1.0.1", "edge/hotfix"].map(
+        channel => parseChannel(channel)
+      );
+    });
+
+    it("should return latest tracks with different branches on the tracks", () => {
+      expect(createChannelTree(channelList)).toEqual({
+        latest: {
+          name: "latest",
+          risks: {
+            stable: {
+              name: "stable",
+              branches: {
+                _base: {
+                  name: "_base"
+                }
+              }
+            },
+            beta: {
+              name: "beta",
+              branches: {
+                test: {
+                  name: "test"
+                }
+              }
+            },
+            edge: {
+              name: "edge",
+              branches: {
+                hotfix: {
+                  name: "hotfix"
+                },
+                "1.0.1": {
+                  name: "1.0.1"
+                }
+              }
+            }
+          }
+        }
+      });
+    });
+  });
+  describe("mix of risk/branch and risk only", () => {
+    beforeEach(() => {
+      channelList = [
+        "stable",
+        "stable/test",
+        "stable/1.0.1",
+        "stable/hotfix"
+      ].map(channel => parseChannel(channel));
+    });
+    it("should return no branch risk, above all branches", () => {
+      expect(createChannelTree(channelList)).toEqual({
+        latest: {
+          name: "latest",
+          risks: {
+            stable: {
+              name: "stable",
+              branches: {
+                _base: {
+                  name: "_base"
+                },
+                test: {
+                  name: "test"
+                },
+                "1.0.1": {
+                  name: "1.0.1"
+                },
+                hotfix: {
+                  name: "hotfix"
+                }
+              }
+            }
+          }
+        }
+      });
+    });
+  });
+});
+
+describe("sortAlphaNum", () => {
+  it("sorts numbers", () => {
+    expect(sortAlphaNum(["0", "5", "2", "6", "7", "1", "8"])).toEqual([
```

Comment:
I guess this should work, but would be good to test how "10" is sorted against smaller numbers (to check if it's not sorted alphabetically).